library/Exceptions/ResourceTypeException.php

```diff
@@ -14,12 +14,14 @@
 namespace Respect\Validation\Exceptions;
 
 /**
+ * Exception class for ResourceType.
+ *
  * @author Henrique Moody <henriquemoody@gmail.com>
  */
-class ResourceTypeException extends ValidationException
+final class ResourceTypeException extends ValidationException
 {
     /**
-     * @var array
+     * {@inheritdoc}
```

Comment:
maybe better use `{{name}} must be of type resource` to keep consistent with the others?

Reply:
I kept it because for the others make sense to have "type" and "value", but for resources only "type" makes sense, so keeping as is sounds better to me. Same will happen to callable, null, iterable...

Reply:
I see what you mean.

For resource and null is ok. But for `iterable`, `callable` and `object` (in 7.2), since we can type hint them, it might make sense to call them type indeed.

Reply:
Good point.